src/parser/mod.rs

```diff
@@ -6926,14 +6927,16 @@ impl<'a> Parser<'a> {
         let comment = if self.parse_keyword(Keyword::COMMENT) {
             let has_eq = self.consume_token(&Token::Eq);
             let next_token = self.next_token();
-            match next_token.token {
-                Token::SingleQuotedString(str) => Some(if has_eq {
-                    CommentDef::WithEq(str)
-                } else {
-                    CommentDef::WithoutEq(str)
-                }),
+            let comment = match next_token.token {
+                Token::SingleQuotedString(str) => str,
+                Token::DollarQuotedString(str) => str.value,
                 _ => self.expected("comment", next_token)?,
-            }
+            };
```

Comment:
can we move this part to be reusable? something like
```rust
fn parse_comment_value() -> Result<String> {
          let value = match next_token.token {
                Token::SingleQuotedString(str) => str,
                Token::DollarQuotedString(str) => str.value,
                _ => self.expected("comment", next_token)?,
            };
         Ok(value)
}
```
and then to call the function from both places

Reply:
Updated.